eth/client.go

```diff
@@ -738,8 +739,99 @@ func (c *client) Vote(pollAddr ethcommon.Address, choiceID *big.Int) (*types.Tra
 	return poll.Vote(opts, choiceID)
 }
 
+func (c *client) Reward() (*types.Transaction, error) {
+	addr := c.accountManager.Account().Address
+
+	t, err := c.GetTranscoder(addr)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder %v", addr.Hex())
+	}
+
+	mintable, err := c.CurrentMintableTokens()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get curernt mintable tokens")
+	}
+
+	totalBonded, err := c.GetTotalBonded()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get total bonded")
+	}
+
+	// reward = (current mintable tokens for the round * active transcoder stake) / total active stake
+	reward := new(big.Int).Div(new(big.Int).Mul(mintable, t.DelegatedStake), totalBonded)
+
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get transcoder pool")
+	}
+
+	hints := c.getStakingHints(addr, ethcommon.Address{}, reward, transcoders)
+	return c.RewardWithHint(hints.newPosPrev, hints.newPosNext)
+
+}
+
 // Helpers
 
+// getStakingHints returns hints for both the new and old (if applicable) delegate
+// amount provided should be negative when unbonding.
+func (c *client) getStakingHints(newDel ethcommon.Address, oldDel ethcommon.Address, amount *big.Int, transcoders []*lpTypes.Transcoder) *struct {
```

Comment:
This function also does not seem to be able to handle the case where the new delegate's stake increases by X and the old delegate's stake decreases by Y where X > Y (i.e. if a delegator moves stake *and* bonds additional stake).

Additionally, having to pass an additional empty address argument feels cumbersome since it would be required in most cases except for when a delegator is moving stake from an old delegate to a new delegate.

WDYT about something like the following:

```
func simulateTranscoderPoolUpdate(del ethcommon.Address, newStake *big.Int, transcoders []*lpTypes.Transcoder) ... {
    for _, t := range transcoders {
        if t.Address == del {
            // Update delegated stake
        }
    }

    // Sort transcoders

    // Calculate and return hint for `del`
}
```

Usage for the case mentioned above would be something like:

```
oldDelStake = oldDel.stake
oldDelHint = simulateTranscoderPoolUpdate(oldDel, oldDelStake, transcoders)

newDelStake = newDel.stake
// transcoders would already be updated based on oldDel update
newDelHint = simulateTranscoderPoolUpdate(newDel, newDelStake, transcoders)
```

The downside of this approach is that two calls to `simulateTranscoderPoolUpdate` would require two sort operations. This would only be the case when using this function for a delegator that is moving stake. Doesn't feel pressing to address right now, but if this becomes a problem we could always remove the sort operation and instead do a linear search + insertion - the transcoder pool would start off sorted anyway so each linear search + insertion would result in a slice that is still sorted.